### PR TITLE
[SYSE-382] Refactor test-square

### PR DESCRIPTION
**Pull request by alephnull**
Unified test-square-api and test-square-ui to test-square.
New v2 API for tui which controlls ui, api and upgrade tests with multiple sources.
v2 api source can be chosen with `env.VARIATION` at the top-level.
Do not build tat.
RHEL multiarch upgrade tests.
pre-commit hook in .githooks, use with `git config --local core.hooksPath .githooks`.
update-action-versions make target.

### CODE REVIEW

File: policy/templates/subtemplates/auto/auto-test.gotmpl [hunk 1]
```diff
@@ -19,65 +14,55 @@
...
           - pump: $ECR/tyk-pump:master
             sink: tykio/tyk-mdcb-docker:v2.4
         {{- end }}
     steps:
+      {{- if eq .dot.Name "tyk-analytics" }}
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          sparse-checkout: tests/{{ .test }}
+      {{- end }}
```
Comment (ermirizio) [inline policy/templates/subtemplates/auto/auto-test.gotmpl:65]: @alephnull test/ui folder is always needed for repos in order to run UI tests, we need to change this conditional to be based on test name instead

File: policy/templates/subtemplates/auto/auto-test.gotmpl [hunk 2]
```diff
@@ -152,71 +107,70 @@
...
+          {{- if eq .dot.Name "tyk-pro" }}
+          BASE_REF: master
+          {{- else }}
           BASE_REF: {{`${{startsWith(github.event_name, 'pull_request') && github.base_ref || github.ref_name }}`}}
-        {{- end }}
+          {{- end }}
         run: |
           # Generate report id
           echo "id=$(date +%s%N)" >> $GITHUB_OUTPUT
+          mkdir /app && ln -s test/api/pytest_ci.ini /app/pytest_ci.ini
```
Comment (ermirizio) [inline policy/templates/subtemplates/auto/auto-test.gotmpl:159]: @alephnull is not possible to write over /app on github runner due to permission error. We may need to decouple or add if statement to diferentiate between local and inside container pytest command

File: policy/templates/subtemplates/auto/auto-test.gotmpl [hunk 3]
```diff
@@ -152,71 +107,74 @@
...
+          BASE_REF: master
+          {{- else }}
           BASE_REF: {{`${{startsWith(github.event_name, 'pull_request') && github.base_ref || github.ref_name }}`}}
-        {{- end }}
+          {{- end }}
         run: |
           # Generate report id
           echo "id=$(date +%s%N)" >> $GITHUB_OUTPUT
+          {{- if eq .dot.Name "tyk-analytics" }}
+          pytest_file="{{`${{ github.workspace }}`}}/tests/{{ .test }}/pytest_local.ini"
```
Comment (alephnull) [inline policy/templates/subtemplates/auto/auto-test.gotmpl:160]: Why pytest_local.ini here? Should it not be pytest_ci.ini always?

File: policy/templates/subtemplates/auto/auto-test.gotmpl (hunk 1 above)
Comment (alephnull) [reply policy/templates/subtemplates/auto/auto-test.gotmpl]: It is always available now.

File: policy/templates/subtemplates/auto/auto-test.gotmpl (hunk 2 above)
Comment (alephnull) [reply policy/templates/subtemplates/auto/auto-test.gotmpl]: I changed your implementation to make it self-contained within the templates.

**Review by ermirizio — APPROVED** [review]
I have reviewed , this is good to merge, lets continue the work on another PR

**Pull request merged**